### PR TITLE
Add option to control airmass/perez enhancement factor in clearsky.ineichen

### LINKED ISSUE #435: Artefact in Ineichen clearsky model for sun close to horizon
Consider following example:
```python
import pandas as pd
import pvlib
times = pd.date_range('2018-02-16', '2018-02-17', freq='min')
location = pvlib.location.Location(latitude=60., longitude=0.)
for model in ['ineichen', 'haurwitz', 'simplified_solis']:
    location.get_clearsky(times, model=model)['ghi'].plot(label=model, legend=True)
```
which results in following figure:
pvlib_clearsky

The (default) 'ineichen' clearsky model seems erratic at sunrise and sunset. The effect is especially visible for high latitudes during Winter when the sun slowly approaches the horizon.

Upon closer look in the code, I found the following comment:
https://github.com/pvlib/pvlib-python/blob/04bd7532f54b550313d84309c3d3d19818694ccc/pvlib/clearsky.py#L98-L103
which refers to the formula:
https://github.com/pvlib/pvlib-python/blob/04bd7532f54b550313d84309c3d3d19818694ccc/pvlib/clearsky.py#L122-L126

However, apart from the typo mentioned, there is another important difference between both formula of Solar Energy 73, and that's the `np.exp(0.01*airmass_absolute**1.8)` factor, which is only present in the formula of pg 311, but not in the formula of pg 156!

It seems like the artefact is exactly a result of that `np.exp(0.01*airmass_absolute**1.8)` factor. This factor slowly increases from 1.01 at solar zenith to 1.95 at a zenith angle of 85°, but then shoots to 1000 when the sun reaches the horizon.

So, maybe there was an error also in the formula of Solar Energy 73, pg 311, and that factor wasn't supposed to be there?

Nevertheless, in reference [3] (Reno et al., 2012), this model was found to excel for US sites with the same formula as used by pvlib. The US sites were all below the [49th parallel](https://en.wikipedia.org/wiki/49th_parallel_north). I wonder whether this model is still recommendable for higher latitudes?

### PR DESCRIPTION
**Pull request by wholmgren**
See #435 for context.

 - [x] Closes #435
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests must pass on the TravisCI and Appveyor testing services.
 - [x] Code quality and style is sufficient. Passes ``git diff upstream/master -u -- "*.py" | flake8 --diff`` and/or landscape.io linting service.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.

Examples of ineichen with and without the enhancement at a few latitudes, with simplied solis as a reference:

ineichen_perez_32 0
ineichen_perez_60 0
ineichen_perez_70 0

While the differences at higher latitudes are more noticeable, the differences at lower latitudes are still notable: a few % difference at solar noon, unphysical pedestals near the horizon.

Many tests failed when I changed the default, so still work to do on that.

### CODE REVIEW

**Comment by wholmgren** [comment]
Some test failures are due to #481.

**Comment by wholmgren** [comment]
This PR is going to end up accomplishing most of the remaining work for #394. 

To do includes:

- [ ] decouple ModelChain tests from clear sky calculation. would like to use new fixtures proposed in #491 
- [ ] update values in ``test_ineichen*`` failures
- [ ] figure out``detect_clearsky`` failures

**Comment by wholmgren** [comment]
I think this is close to ready to go. It needs someone to review it, especially the changes to ``clearsky.py`` and ``test_clearsky.py``.

The majority of the other changes are cleaning up the test suite so that it's not tightly coupled to this function. The ``perez_enhancement`` change made all of those test functions fail so I think we had no choice but to address the broader changes here.

The ``detect_clearsky`` failures were caused by the scaling factor, ``alpha``, changing in response to the ~1% change in ghi returned by the ``ineichen`` function. I adjusted the expected test values to get them to pass.

**Comment by adriesse** [comment]
For what it's worth, I browsed through the changes and didn't notice any problems.

Is it possible to have tests that compare outputs to the previous stable release, instead of hard coding lists of constants?

**Comment by wholmgren** [comment]
Thanks @adriesse. Interesting idea to compare results to previous stable releases. Maybe there is a framework/package for that?

**Comment by wholmgren** [comment]
@cwhanse I changed the comments. I'm not 100% sure I captured what you were going for.

I think this is ready to merge. The test pass on all Travis CI configurations. The appveyor test failures are intermittent and unrelated.

**Comment by cwhanse** [comment]
I think the comments are fine.  My intent was to discard the no-longer relevant text from the Matlab file, and to clarify how this code implements the equations in those two Ineichen/Perez papers.

Any comments from others before this is merged?

File: pvlib/test/test_clearsky.py [hunk 1]
```diff
@@ -20,35 +21,66 @@
...
-    am = atmosphere.relativeairmass(ephem_data['apparent_zenith'])
-    am = atmosphere.absoluteairmass(am, atmosphere.alt2pres(tus.altitude))
+    times = pd.date_range(start='2014-06-24', end='2014-06-25', freq='3h',
+                          tz='America/Phoenix')
+    apparent_zenith = pd.Series(np.array(
+        [124.0390863, 113.38779941, 82.85457044, 46.0467599, 10.56413562,
+         34.86074109, 72.41687122, 105.69538659, 124.05614124]),
+        index=times)
+    am = pd.Series(np.array(
+        [nan, nan, 6.97935524, 1.32355476, 0.93527685,
```
[the remark this answers is not in the document]
Comment (wholmgren) [reply pvlib/test/test_clearsky.py]: These arrays are copy/pasted from the ``.values`` attributes of the associated Series/DataFrames. They are printed with ``nan``, not ``np.nan``. Is there a better, easy way to do it?

[the remark this answers is not in the document]
Comment (wholmgren) [reply pvlib/test/test_clearsky.py]: I totally agree with you in general, just doing this for expediency in test code.

[the remark this answers is not in the document]
Comment (mikofski) [reply pvlib/test/test_clearsky.py]: OK, I see. I can't think of any easy way to deal with that, other than maybe find and replace?

**Pull request merged**